File: LEETCODE/Number_of_1_Bits/solution.ts
function hammingWeight(n: number): number {
  const bin = n.toString(2);
  let count = 0;
  for (let bit of bin) {
    if (bit === "1") count++;
  }
  return count;
}

function hammingWeight2(n: number): number {
  let count = 0;
  while (n !== 0) {
    n &= n - 1;
    count++;
  }
  return count;
}

File: LEETCODE/Number_of_1_Bits/solution.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        return bin(n).count("1")

    def hammingWeight2(self, n: int) -> int:
        count = 0

        while n != 0:
            n &= n-1
            count += 1

        return count
